app/src/main/java/protect/card_locker/LoyaltyCardCursorAdapter.java

```diff
@@ -335,20 +339,41 @@ public void toggleStar(boolean enable, boolean colorByTheme) {
             if (colorByTheme) {
                 dark = !mDarkModeEnabled;
             }
+
             if (dark) {
                 mStarBorder.setImageResource(R.drawable.ic_unstarred_white);
                 mStarBackground.setImageResource(R.drawable.ic_starred_black);
             } else {
                 mStarBorder.setImageResource(R.drawable.ic_unstarred_black);
                 mStarBackground.setImageResource(R.drawable.ic_starred_white);
             }
-            if (enable) {
+
+            if (enableStar) {
                 mStar.setVisibility(View.VISIBLE);
             } else {
                 mStar.setVisibility(View.GONE);
             }
+
             mStarBorder.invalidate();
             mStarBackground.invalidate();
+
+            if (dark) {
+                mArchivedBorder.setImageResource(R.drawable.ic_baseline_archive_24);
+                mArchivedBackground.setImageResource(R.drawable.ic_baseline_archive_24_black);
+            } else {
+                mArchivedBorder.setImageResource(R.drawable.ic_baseline_archive_24_black);
+                mArchivedBackground.setImageResource(R.drawable.ic_baseline_archive_24);
+            }
+
+            if (enableArchive) {
+                mArchived.setVisibility(View.VISIBLE);
+            } else {
+                mArchived.setVisibility(View.GONE);
+            }
+
+            mArchivedBorder.invalidate();
+            mArchivedBackground.invalidate();
```

Comment:
I'm not a huge fan of:

```
if (dark) {
  A
} else {
  B
}

C.invalidate()

if (dark) {
  D 
} else {
  E
}

F.invalidate()
```

I'd find it much simpler to use:
```
if (dark) {
  A
  D
} else {
  B
  E
}

C.invalidate()
F.invalidate()
```

Although this is kinda a nitpick